api/api_logging.go

```diff
@@ -120,25 +121,30 @@ func encryptPassword(buf []byte) []byte {
 		separator := ""
 		l = sc.Text()
 
-		switch {
-		case strings.Contains(l, "password"):
-			match = `"password":"`
-			separator = `"`
-		case strings.Contains(l, "Cookie: isisessid="):
-			match = `isisessid=`
-			separator = `-`
-		case strings.Contains(l, "X-Csrf-Token"):
-			match = `X-Csrf-Token:`
-			separator = `-`
-		case strings.Contains(l, "Authorization: Basic"):
-			match = `Basic `
-		}
-		if match != "" {
-			startIndex, endIndex, matchStrLen := FetchValueIndexForKey(l, match, separator)
-			if startIndex >= 0 && endIndex > 0 { // if the separator is present then replace only the characters till separator with the special character
-				l = l[:startIndex+matchStrLen] + "****" + l[startIndex+matchStrLen+endIndex:]
-			} else if startIndex >= 0 { // if the separator in not present then replace the string to be masked with the special character
-				l = l[:startIndex+matchStrLen] + "****"
+		if strings.Contains(l, "Authorization: Basic") {
+			base64str := strings.Split(l, " ")[2]
```

Comment:
I hope we are sure that this index 2 will be present, else will lead to panic

Reply:
the exact statement is like this _Authorization: Basic cmBYJW5jXYZ_ where the first 2 are request format and the third part is base64 encoded username and password